### DISCUSSION TITLE
Specify location of local maven repo

### DISCUSSION THREAD

**Discussion by jrtct** [aquasecurity/trivy#6679] [Ideas]
### Description

I'm working for a client that uses a lot of Java applications that are built with maven. We're noticing trivy can be quite slow when scanning a project with a lot of dependencies. A way to speed this up would be to build the project first and then scan the local repository (`~/.m2/repository`). We're already caching this directory, but the problem is trivy does not find it. This seems to be because trivy expects it to be in $HOME, which is not necessarily the case, especially in a CI context. If there would be a way to specify the location of this local repository as an option, this would greatly speed up scans in our case.

Maybe I'm overlooking something and I'm just using trivy wrong, but I didn't really see any other solution

Looking forward to your reply!

### Target

Filesystem

### Scanner

Vulnerability

**Comment by nikpivkin** [comment]
Hi @jrtct !

Trivy respects the localRepository from settings.xml. https://maven.apache.org/settings.html#simple-values

**Reply by jrtct** [reply]
Hi @nikpivkin,
thanks very much for that tip! In our case we're already building a custom trivy image anyway (to add a plugin). So adding a custom settings.xml into the trivy container is easy to do.
I would have never thought of this, so thank you very much for pointing me in the right direction!
